first-party/jib-layer-filter-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/layerfilter/JibLayerFilterExtension.java

```diff
@@ -102,9 +122,120 @@ public ContainerBuildPlan extendContainerBuildPlan(
         .map(FileEntriesLayer.Builder::build)
         .filter(layer -> !layer.getEntries().isEmpty())
         .forEach(newPlanBuilder::addLayer);
+
+    ContainerBuildPlan newPlan = newPlanBuilder.build();
+
+    if (config.get().isCreateParentLayers()) {
+      newPlan = moveParentDepsToNewLayers(newPlan, mavenData, logger);
+    }
+
+    return newPlan;
+  }
+
+  private ContainerBuildPlan moveParentDepsToNewLayers(
+      ContainerBuildPlan buildPlan, MavenData mavenData, ExtensionLogger logger)
+      throws JibPluginExtensionException {
+
+    logger.log(LogLevel.LIFECYCLE, "Moving parent dependencies to new layers.");
+
+    // the key is the expected path for the parent dependency
+    Map<String, Artifact> parentDependencies =
+        getParentDependencies(mavenData)
+            .stream()
+            .map(d -> d.getArtifact())
```

Comment:
Sorry, this repo style doesn't allow a single-letter name 'd'.
```suggestion
            .map(dep -> dep.getArtifact())
```
or we usually do
```suggestion
            .map(Dependency::getArtifact)
```

Reply:
Ok. And I tried to fix other occurences, too. Unfortunately, checkstyle does not complain...